akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -512,6 +520,20 @@ private[stream] class ActorGraphInterpreter(_initial: GraphInterpreterShell) ext
         false
     }
 
+  //this limits number of messages that can be processed synchronously during one actor receive.
+  private val shortCircuitLimit: Int = _initial.mat.settings.syncProcessingLimit
```

Comment:
I'm afraid that the default 32 is too short for such a global limit. In the case of HTTP there are 20 something stages fused, so those can be 40 events worst case for a full push-pull cycle. Probably less, but it feels still dangerously close to the "edge".